HLTrigger/Timer/plugins/FastTimerService.cc

```diff
@@ -1662,17 +1665,61 @@ void FastTimerService::postModuleStreamEndLumi(edm::StreamContext const& sc, edm
   thread().measure_and_accumulate(lumi_transition_[index]);
 }
 
-void FastTimerService::on_scheduler_entry(bool worker) {
-  // initialise the measurement point for a thread that has newly joining the TBB pool
-  thread().measure();
+FastTimerService::ThreadGuard::ThreadGuard() {
+  auto err = ::pthread_key_create(&key_, retire_thread);
```

Comment:
If I understand correctly, `retire_thread` is called when the worker thread exits.
Why not call it from `on_scheduler_exit` instead ?

Reply:
Monitoring the primary arena, `on_scheduler_entry` and `on_scheduler_exit` get called many times as threads get moved between different arenas, and there's no way to tell when the call to `on_scheduler_exit` is the final exit.  This is the only reliable way I could think of to catch if a thread gets deleted before the end job sequence.

Reply:
OK, so with this approach, the time (and resources) spent by a thread *outside* the main arena would be accounted as "overhead", right ?
Which is not a bad thing.

Reply:
I believe that with either the global observer or this PR, what gets measured as overhead is the time outside of any defined CMS module, irrespective of the arena.  With the global observer you were getting one call to `on_scheduler_enter` and `on_scheduler_exit` per thread, and I'm trying to replicate that with the observer on the primary arena.